libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -782,6 +788,108 @@ def from_documents(
+        retrieval_query: str = "",
+        **kwargs: Any,
+    ) -> Neo4jVector:
+        """
+        Return Neo4jVector initialized from existing graph.
+        Neo4j credentials are required in the form of `url`, `username`,
+        and `password` and optional `database` parameters.
+        """
+
+        # Prefer retrieval query from params, otherwise construct it
+        if not retrieval_query:
+            retrieval_query = (
+                f"RETURN reduce(str='', k IN {text_node_properties} |"
+                " str + '\\n' + k + ':' + coalesce(node[k], '')) AS text, "
+                "node {.*, `"
+                + embedding_node_property
+                + "`: Null, id: Null, "
+                + ", ".join([f"`{prop}`: Null" for prop in text_node_properties])
+                + "} AS metadata, score"
+            )
+        store = cls(
+            embedding=embedding,
+            index_name=index_name,
+            keyword_index_name=keyword_index_name,
+            search_type=search_type,
+            retrieval_query=retrieval_query,
+            node_label=node_label,
+            embedding_node_property=embedding_node_property,
+            **kwargs,
+        )
+
+        # Check if the vector index already exists
+        embedding_dimension = store.retrieve_existing_index()
+
+        # If the vector index doesn't exist yet
+        if not embedding_dimension:
+            store.create_new_index()
+        # If the index already exists, check if embedding dimensions match
+        elif not store.embedding_dimension == embedding_dimension:
+            raise ValueError(
+                f"Index with name {store.index_name} already exists."
+                "The provided embedding function and vector index "
+                "dimensions do not match.\n"
+                f"Embedding function dimension: {store.embedding_dimension}\n"
+                f"Vector index dimension: {embedding_dimension}"
+            )
+        # FTS index for Hybrid search
+        if search_type == SearchType.HYBRID:
+            fts_node_label = store.retrieve_existing_fts_index(text_node_properties)
+            # If the FTS index doesn't exist yet
+            if not fts_node_label:
+                store.create_new_keyword_index(text_node_properties)
+            else:  # Validate that FTS and Vector index use the same information
+                if not fts_node_label == store.node_label:
+                    raise ValueError(
+                        "Vector and keyword index don't index the same node label"
+                    )
+
+        # Populate embeddings
+        while True:
+            fetch_query = (
+                f"MATCH (n:{node_label}) "
+                f"WHERE n.{embedding_node_property} IS null "
+                "AND any(k in $props WHERE n[k] IS NOT null) "
+                f"RETURN id(n) AS id, reduce("
+                "str='', k IN $props | str + '\\n' + k + ':' + n[k]) AS text "
```

Comment:
coalesce(n[k]) see the retrieval query above, 
can we reuse the expression
space after colon

we can also filter the non-null ones?